libyul/AsmParser.cpp

```diff
@@ -78,6 +64,65 @@ unique_ptr<Block> Parser::parse(std::shared_ptr<Scanner> const& _scanner, bool _
 	return nullptr;
 }
 
+langutil::Token Parser::advance()
+{
+	auto const token = ParserBase::advance();
+	updateLocation();
+	return token;
+}
+
+void Parser::updateLocation()
+{
+	if (m_scanner->currentCommentLiteral().empty())
+		return;
+
+	// parse doxygen style comment with a sequence of key/value pairs
+	// TODO: [ ] docflag!
+	// TODO: [x] join regexes into one, hard search for @src
+	// TODO: [x] catch exception on stoi (e.g. too biig number?)
+	// TODO: [ ] updateLocationEndFrom(): pull top if-condition out of it
+
+	static std::regex const lineRE = std::regex(
+		R"~~~(.*@src\s+(\d+):(-1|\d+):(-1|\d+).*)~~~",
```

Comment:
I think `.*` is not useful because it would also parse `x@src`

Reply:
changed vom `.*` to `\s*`.